qokit/qaoa_circuit.py

```diff
@@ -52,12 +51,13 @@ def append_x_term(qc: QuantumCircuit, q1: int, beta: float) -> None:
 def append_cost_operator_circuit(qc: QuantumCircuit, terms: Sequence, gamma: float) -> None:
     for term in terms:
         if len(term) == 2 and isinstance(term[1], tuple):
-            coeff, (i, j) = term
-            append_z_prod_term(qc, (i, j), gamma * coeff / 2)
+            coeff, term_tuple = term
+            append_z_prod_term(qc, term_tuple, gamma * coeff / 2)
         elif any([isinstance(i, tuple) for i in term]):
             raise ValueError(f"Invalid term received: {term}")
         else:
-            append_z_prod_term(qc, term, gamma)
+            term_weight = len(term)
+            append_z_prod_term(qc, term, gamma * term_weight / 2)
```

Comment:
Why do we do this? Terms with coefficients should already be handled by the first if statement. We don't want to hardcode the term length as the coefficient in case there are other problems with four-body terms.

Reply:
I maybe missing something here, but what would the coefficient in case of other problems with four-body term. This was done to handle cases where the terms are passed only as a tuple not including the coefficients such as in this test: https://github.com/jpmorganchase/QOKit/blob/7ffb97292a83a85a4af935a0d9188dc661d3970f/tests/test_fast_simulators_labs.py#L93

Reply:
Removed the hardcoded `len(terms)`.

Reply:
Per the docstring, terms without coefficients should be treated as having coefficients 1:
https://github.com/jpmorganchase/QOKit/blob/9d8e64e490b2cb24885aa805e9c4a625d8140125/qokit/qaoa_circuit.py#L83-L85